src/driftpy/dlob/DLOB.py

```diff
@@ -0,0 +1,1067 @@
+                nodes_to_fill.append(NodeToFill(node, []))
+
+        return nodes_to_fill
+    
+    def find_taking_nodes_crossing_maker_nodes(
+        self,
+        market_index: int,
+        slot: int,
+        market_type: MarketType,
+        oracle_price_data: OraclePriceData,
+        taker_node_generator: Generator[DLOBNode, None, None],
+        maker_node_generator_fn: Callable[[int, int, MarketType, OraclePriceData], Generator[DLOBNode, None, None]],
+        does_cross: Callable[[int, int], bool]
+    ) -> List[NodeToFill]:
+        nodes_to_fill: List[NodeToFill] = []
+
+        for taker_node in taker_node_generator:
+            maker_node_generator = maker_node_generator_fn(market_index, slot, market_type, oracle_price_data)
+
+            for maker_node in maker_node_generator:
+                # Check if nodes are from the same user
+                if taker_node.user_account == maker_node.user_account:
+                    continue
+
+                maker_price = maker_node.get_price(oracle_price_data, slot)
+                taker_price = taker_node.get_price(oracle_price_data, slot)
+
+                if not does_cross(taker_price, maker_price):
+                    break
+
+                nodes_to_fill.append(NodeToFill(taker_node, [maker_node]))
+
+                # Update orders
+                maker_base_remaining = maker_node.order.base_asset_amount - maker_node.order.base_asset_amount_filled
+                taker_base_remaining = taker_node.order.base_asset_amount - taker_node.order.base_asset_amount_filled
+
+                base_filled = min(maker_base_remaining, taker_base_remaining)
+
+                new_maker_order = copy.deepcopy(maker_node.order)
+                new_maker_order.base_asset_amount_filled += base_filled
+
+                self.get_list_for_order(new_maker_order, slot).update(new_maker_order, maker_node.user_account)
+
+                new_taker_order = copy.deepcopy(taker_node.order)
+                new_taker_order.base_asset_amount_filled += base_filled
+
+                self.get_list_for_order(new_taker_order, slot).update(new_taker_order, taker_node.user_account)
+
+                if new_taker_order.base_asset_amount_filled == taker_node.order.base_asset_amount:
+                    break
+        
+        return nodes_to_fill
+    
+    def find_jit_auction_nodes_to_fill(
+        self,
+        market_index: int,
+        slot: int,
+        oracle_price_data: OraclePriceData,
+        market_type: MarketType
+    ) -> List[NodeToFill]:
+        nodes_to_fill: List[NodeToFill] =[]
+
+        for market_bid in self.get_taking_bids(market_index, market_type, slot, oracle_price_data):
+            nodes_to_fill.append(NodeToFill(market_bid, []))
+
+        for market_ask in self.get_taking_asks(market_index, market_type, slot, oracle_price_data):
+            nodes_to_fill.append(NodeToFill(market_ask, []))
+    
+    def find_taking_nodes_to_fill(
```

Comment:
should add a comment that this needs unit tests before it should be used in prod ...